vrx_ign/src/ScoringPlugin.hh

```diff
@@ -0,0 +1,291 @@
+
+        /// \brief gazebo node pointer
+        private: std::unique_ptr<ignition::transport::Node> gzNode;
+
+        /// \brief Collision detection node subscriber
+        // TODO - Verify if ignition requires a subscriber object
+        
+        /// \brief gazebo server control publisher
+        std::unique_ptr<ignition::transport::Node::Publisher> serverControlPub;
+
+        /// \brief Topic where the task stats are published.
+        private: std::string taskInfoTopic = "/vrx/task/info";
+
+        /// \brief Bool flag for debug.
+        private: bool debug = true;
+
+        /// \brief Topic where debug collision is published.
+        private: std::string contactDebugTopic = "/vrx/debug/contact";
+
+        /// \brief The score.
+        private: double score = 0.0;
+
+        /// \brief Pointer to the SDF plugin element.
+        private: std::shared_ptr<const sdf::Element> sdf; 
+
+        /// \brief Duration (seconds) of the initial state.
+        private: double initialStateDuration = 30.0;
+
+        /// \brief Duration (seconds) of the ready state.
+        private: double readyStateDuration = 60.0;
+
+        /// \brief Duration (seconds) of the running state (max task time).
+        protected: double runningStateDuration = 300.0;
+
+        /// \brief Absolute time specifying the start of the ready state.
+        private: std::chrono::duration<double> readyTime;
+
+        /// \brief Absolute time specifying the start of the running state.
+        private: std::chrono::duration<double> runningTime;
+
+        /// \brief Absolute time specifying the start of the finish state.
+        private: std::chrono::duration<double> finishTime;
+
+        /// \brief Current time (simulation).
+        private: std::chrono::duration<double> currentTime;
+
+        // \brief Elapsed time since the start of the task (running state).
+        private: std::chrono::duration<double> elapsedTime;
+
+        /// \brief Remaining time since the start of the task (running state).
+        private: std::chrono::duration<double> remainingTime;
+
+        /// \brief Collision buffer.
+        private: float collisionBuffer = 3.0;
+
+        /// \brief Collisions counter.
+        private: int collisionCounter = 0;
+
+        /// \brief Collision list.
+        private: std::vector<std::string> collisionList;
+
+        /// \brief Collisions timestamps.
+        private: std::vector<std::chrono::duration<double> > collisionTimestamps;
+
+        /// \brief Whether the current task has timed out or not.
+        private: bool timedOut = false;
+
+        /// \brief Time at which the last message was sent.
+        private: std::chrono::duration<double> lastStatsSent;
+
+        /// \brief The task state.
+        private: std::string taskState = "initial";
+        
+        /// \brief The next task message to be published.
+        ignition::msgs::Param_V taskMsg;
+        ignition::msgs::Any taskMsgName, taskMsgState, 
```

Comment:
Nit: Following the Gazebo convention, don't group member variables. Instead, declare them in separate lines.